projects/common/src/time/time-duration.ts

```diff
@@ -26,6 +26,25 @@ export class TimeDuration {
     return `PT${this.toMillis() / 1000}S`;
   }
 
+  public toTimeAgoString(): string {
+    const secondsAgo = this.calcSecondsAgo();
+
+    if (secondsAgo < 59) {
+      return 'Just now';
+    }
+
+    const mostSignificantPortion = this.getMostSignificantUnitOnly();
+
+    const value = mostSignificantPortion.value;
+    const units = mostSignificantPortion.toUnitString();
+
```

Comment:
line 38-41 are basically a reimplementation of `toLongString()` I believe. Rather than adding any code in time duration, I'd expect the timeAgo pipe to use duration (illustrated below). Putting this logic into time duration doesn't feel right because a duration has no reference to a point in time - it's only a length of time.